### ISSUE TITLE
Examples trait requires output even in error example

### ISSUE THREAD

**Issue by kubukoz** [awslabs/smithy#1582]
## What's happening

Hi! I think the validator of the examples trait isn't behaving correctly - it seems to require an `output` entry (and a valid one, at that) even if the example exhibits an error.

This doesn't seem intentional, looking at [the example from the PR](https://github.com/awslabs/smithy/pull/888/files#diff-25d3a8f46e169673c09176b5405f96d809ff05c58f5bf302244457b53cd5facbR245-R256) that introduced error examples.

## Reproduction

This can be reproduced by running the following with [scala-cli](https://scala-cli.virtuslab.org/):

```scala
//> using lib "software.amazon.smithy:smithy-model:1.27.1"
import software.amazon.smithy.model.Model

object demo extends App {
  val model = Model
    .assembler()
    .addUnparsedModel(
      "test.smithy",
      """
      $version: "2.0"

      namespace example.hello

      operation Hello {
          output: Greeting,
          errors: [YouShallNotPass]
      }

      structure Greeting {
          @required
          message: String
      }

      @error("client")
      structure YouShallNotPass {
          @required
          message: String
      }

      apply Hello @examples([
          {
              title: "My name is Jeff"
              output: {
                  message: "Hello Jeff"
              }
          },
          {
              title: "Unhappy path",
              error: {
                  shapeId: YouShallNotPass
                  content: {
                      "message": "Go back to the Shadow"
                  }
              }
          }
      ])
      """
    )
    .assemble()
    .unwrap()

  println("success")
}
```

Here's what you get:

```
software.amazon.smithy.model.validation.ValidatedResultException: Result contained ERROR severity validation events: 
[ERROR] example.hello#Hello: Example output of `Unhappy path`: Missing required structure member `message` for `example.hello#Greeting` | ExamplesTrait N/A:0:0
        at software.amazon.smithy.model.validation.ValidatedResult.validate(ValidatedResult.java:145)
        at software.amazon.smithy.model.validation.ValidatedResult.unwrap(ValidatedResult.java:132)
        at demo$.<clinit>(main.scala:9)
```

## Why I think this happens

I believe this code should consider `output` and `error` mutually exclusive, but it seems to always validate `output`:

https://github.com/awslabs/smithy/blob/f5a0893cd0f9983d0bd5697db7c226c1d379ef92/smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ExamplesTraitValidator.java#L56-L75

## Extras

I think the [documentation](https://smithy.io/2.0/spec/documentation-traits.html?highlight=examples) is a little broken: this isn't valid Smithy code

<img width="611" alt="image" src="https://user-images.githubusercontent.com/894884/212777787-5b52a909-5017-4225-88a8-bdfbf4a0b3fc.png">

**Comment by jvschneid** [comment]
Hi Jakub,

Thanks for bringing this issue to our attention. I've raised a draft PR to address this case and discuss the approach with our team.

**Comment by jvschneid** [comment]
Hi Jakub,

Change has been merged and release with Smithy 1.28.0 version, so I'm closing this issue. Let us know if everything else is needed.

**Comment by kubukoz** [comment]
Awesome, thank you very much!
